dali/pipeline/data/tensor_vector.cc

```diff
@@ -172,240 +196,301 @@ void CopyImpl(DstBatch<DstBackend> &dst, const SrcBatch<SrcBackend> &src, const
 }  // namespace copy_impl
 
 template <typename Backend>
-TensorVector<Backend>::TensorVector()
-    : curr_num_tensors_(0), tl_(std::make_shared<TensorList<Backend>>()) {}
+TensorVector<Backend>::TensorVector() : curr_num_tensors_(0) {}
 
 
 template <typename Backend>
-TensorVector<Backend>::TensorVector(int batch_size)
-    : curr_num_tensors_(0), tl_(std::make_shared<TensorList<Backend>>(batch_size)) {
+TensorVector<Backend>::TensorVector(int batch_size) : curr_num_tensors_(0) {
+  // We don't use negative batch size through DALI, and by default we wanted batch to
+  // not do any initial allocation unless actual shape is provided.
+  // As the -1 and 0 sample dims (the latter being reserved for scalar case already),
```

Comment:
As the -1 and 0 sample dims (...) [what about them?]

Reply:
Filled in the missing part.